src/msg/xio/XioMessenger.cc

```diff
@@ -221,13 +221,13 @@ static string xio_uri_from_entity(const string &type,
   char addr_buf[129];
   string xio_uri;
 
-  switch(addr.addr.ss_family) {
+  switch(addr.get_family()) {
   case AF_INET:
-    host = inet_ntop(AF_INET, &addr.addr4.sin_addr, addr_buf,
+    host = inet_ntop(AF_INET, &addr.u.sin.sin_addr, addr_buf,
```

Comment:
can you use the helper functions instead of accessing the union directly? `addr.u.sin` -> `addr.in4_addr()`